src/poetry_plugin_export/command.py

```diff
@@ -86,16 +87,27 @@ def handle(self) -> int:
             )
 
         # Checking extras
-        extras = {
-            canonicalize_name(extra)
-            for extra_opt in self.option("extras")
-            for extra in extra_opt.split()
-        }
-        invalid_extras = extras - self.poetry.package.extras.keys()
-        if invalid_extras:
-            raise ValueError(
-                f"Extra [{', '.join(sorted(invalid_extras))}] is not specified."
+        if self.option("extras") and self.option("all-extras"):
+            self.line_error(
+                "<error>You cannot specify explicit"
+                " `<fg=yellow;options=bold>--extras</>` while exporting"
+                " using `<fg=yellow;options=bold>--all-extras</>`.</error>"
             )
+            return 1
+
+        if self.option("all-extras"):
+            extras = self.poetry.package.extras.keys()
+        else:
+            extras = {
+                canonicalize_name(extra)
+                for extra_opt in self.option("extras")
+                for extra in extra_opt.split()
+            }
```

Comment:
mypy is failing here. Maybe, it's enough to insert `extras: Iterable[NormalizedName]` before the `if`.